src/actions/measurements.ts

```diff
@@ -517,14 +577,176 @@ export const changeMeasurementsDisplay = (
 export const changeMeasurementsGroupBy = (
   newGroupBy: string
 ): ThunkFunction => (dispatch, getState) => {
-  const { measurements } = getState();
-  const controlKey = "measurementsGroupBy";
-  const newControls = { [controlKey]: newGroupBy };
+  const { controls, measurements } = getState();
+  const groupingValues = measurements.collectionToDisplay.groupings.get(newGroupBy).values || [];
+  const newControls: Partial<MeasurementsControlState> = {
+    /* If the measurementsColorGrouping is no longer valid, then set to undefined */
+    measurementsColorGrouping: groupingValues.includes(controls.measurementsColorGrouping)
+      ? controls.measurementsColorGrouping
+      : undefined,
+    measurementsGroupBy: newGroupBy
+  };
 
-  dispatch({
-    type: CHANGE_MEASUREMENTS_GROUP_BY,
-    controls: newControls,
-    queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+  batch(() => {
+    dispatch({
+      type: CHANGE_MEASUREMENTS_GROUP_BY,
+      controls: newControls,
+      queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+    });
+
+    /* After the group by has been updated, update the measurement coloring data if needed */
+    updateMeasurementsColorData(
+      newControls.measurementsColorGrouping,
+      controls.measurementsColorGrouping,
+      controls.colorBy,
+      controls.defaults.colorBy,
+      dispatch
+    );
+  })
+}
+
+export function getActiveMeasurementFilters(
+  filters: MeasurementFilters
+): {string?: string[]} {
+  // Find active filters to filter measurements
+  const activeFilters: {string?: string[]} = {};
+  Object.entries(filters).forEach(([field, valuesMap]) => {
+    activeFilters[field] = activeFilters[field] || [];
+    valuesMap.forEach(({active}, fieldValue) => {
+      // Save array of active values for the field filter
+      if (active) activeFilters[field].push(fieldValue);
```

Comment:
i know it's widespread in the codebase but unbraced conditionals always make me twitch, just a little bit. 

just needed to get that off my chest in the new year.

Reply:
Ah yeah...it doesn't bother me as much :face_with_peeking_eye:

Reply:
😆

Reply:
> i know it's widespread in the codebase but unbraced conditionals always make me twitch, just a little bit.

Would you prefer it if it was:

```js
activeFilters[field].push(fieldValue) if active;
```

;-)

Reply:
I was literally considering tooting something, earlier today, about how the loss of postfix conditionals was one of the biggest losses in moving away from Perl. :lolsob: